marxs/source/labSource.py

```diff
@@ -0,0 +1,98 @@
+import numpy as np
+from astropy.table import Table, Column
+
+from .source import Source
+
+
+class FarLabConstantPointSource(Source):
+	'''Simple in-lab source for testing purposes
+
+	- point source
+	- TODO: figure out how to provide energy distribution
+	- photons uniformly distributed through rectangular space (reasonable approximation if source is far)
+	- photon direction determined by location of source, and selected photon starting position
+	'''
+	def __init__(self, position, polarization, rate, energyMean, energyStdDev):
+		self.pos = position
+		self.polar = polarization
+		self.rate = rate   # photons/sec
+		self.meanEnergy = energyMean
+		self.stdDevEnergy = energyStdDev
+
+	def generate_photons(self, t, center, v_y, v_z):
+		n = (int)(t * self.rate)
+		#out = np.empty((6, n))
+		#out[0, :] = np.arange(0, t, 1. / self.rate)
+		# out[1, :] = self.pos
+
+		# randomly choose direction - photons uniformly distributed over baffle plate area
+		# coordinate axes: origin at baffle plate, tube center. +x source, +y window, +z up
+		# measurements in mm
+		y = np.linalg.norm(v_y)
+		z = np.linalg.norm(v_z)
+		pos = np.empty((4, n))
+		pos = np.array([center[0] * np.ones(n),
+						center[1] + np.random.uniform(-y, y, n),
+						center[2] + np.random.uniform(-z, z, n),
+						np.ones(n)])
+		dir = np.empty((4, n))
```

Comment:
You don't need this line. You define `pos` again in the line below anyway. You only need to make the array this way (or with the the very similar `np.zeros` or `np.ones`) if you fill the individual array elements in a for loop.